dojo/importers/base_importer.py

```diff
@@ -0,0 +1,926 @@
+        if finding_count > 0:
+            # Set the base message to indicate how many findings were parsed from the report
+            message = f"{scan_type} processed a total of {finding_count} findings"
+            if import_type == Test_Import.IMPORT_TYPE:
+                # Check for close old findings context to determine if more detail should be added
+                if kwargs.get("close_old_findings", False):
+                    message += f" and closed {closed_finding_count} findings"
+            if import_type == Test_Import.REIMPORT_TYPE:
+                # Add more details for any status changes recorded
+                if new_finding_count:
+                    message += f" created {new_finding_count} findings"
+                if closed_finding_count:
+                    message += f" closed {closed_finding_count} findings"
+                if reactivated_finding_count:
+                    message += f" reactivated {reactivated_finding_count} findings"
+                if untouched_finding_count:
+                    message += f" did not touch {untouched_finding_count} findings"
+            # Drop a period at the end
+            message += "."
+        else:
+            # Set the message to convey that all findings processed are identical to the last time an import/reimport occurred
+            message = "No findings were added/updated/closed/reactivated as the findings in Defect Dojo are identical to those in the uploaded report."
+
+        return message
+
+    def chunk_objects(
+        self,
+        object_list: List[Finding | Endpoint],
+        chunk_size: int = settings.ASYNC_FINDING_IMPORT_CHUNK_SIZE,
+    ) -> List[List[Finding | Endpoint]]:
+        """
+        Split a single large list into a list of lists of size `chunk_size`.
+        For Example
+        ```
+        >>> chunk_objects([A, B, C, D, E], 2)
+        >>> [[A, B], [B, C], [E]]
+        ```
+        """
+        # Break the list of parsed findings into "chunk_size" lists
+        chunk_list = [list[i:i + chunk_size] for i in range(0, len(list), chunk_size)]
+        logger.debug('IMPORT_SCAN: Split endpoints into ' + str(len(chunk_list)) + ' chunks of ' + str(chunk_size))
+        return chunk_list
+
+    def chunk_endpoints_and_disperse(
+        self,
+        finding: Finding,
+        test: Test,
+        endpoints: List[Endpoint],
+        **kwargs: dict,
+    ) -> None:
+        """
+        Determines whether to asynchronously process endpoints on a finding or not. if so,
+        chunk up the findings to be dispersed into individual celery workers. Otherwise,
+        only use one worker
+        """
+        if settings.ASYNC_FINDING_IMPORT:
+            chunked_list = self.chunk_objects(endpoints)
+            # If there is only one chunk, then do not bother with async
+            if len(chunked_list) < 2:
+                self.add_endpoints_to_unsaved_finding(finding, test, endpoints, sync=True)
+                return []
```

Comment:
```suggestion
                return
```

Reply:
Fixed in an upcoming commit

Reply:
Explicitly returned None at the end of the function instead